app/controllers/sessions_controller.rb

```diff
@@ -1,8 +1,25 @@
 class SessionsController < ApplicationController
+  before_action :load_user, only: %i(create)
+
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
+    auth @user
+  end
+
+  def destroy
+    flash[:success] = t("flash.logout")
+    log_out
+    redirect_to root_url
+  end
+
+  private
+
+  def load_user
+    @user = User.find_by(email: params[:session][:email].downcase)
+  end
+
+  def auth user
     if user&.authenticate(params[:session][:password])
```

Comment:
ko cần `&.` nữa nhé, xử lý user ở trên luôn rồi
mà đã dùng instance user thì line này phả là 
```suggestion
    if @user.authenticate(params[:session][:password])
```